fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -635,6 +640,25 @@ public org.springframework.data.domain.Page<LoanTransactionData> retrieveLoanTra
         return transactionPage.map(loanTransactionMapper::mapLoanTransaction);
     }
 
+    @Override
+    public List<LoanTransactionData> retrieveLoanTransactionsByTypes(Long loanId, Collection<LoanTransactionType> transactionTypes) {
```

Comment:
A simple JPQL query in Repository would have been better... I dont like the idea to fetch LoanTransaction entities and then map to LoanTransactionData.

Lets do DTO projection instead!
Example: org.apache.fineract.organisation.monetary.domain.ApplicationCurrencyRepository#FIND_CURRENCY_DETAILS